src/util/windows.jl

```diff
@@ -0,0 +1,53 @@
+"""
+An iterator which moves across a container, as a sliding window.
+"""
+immutable EachWindowIterator{T <: AbstractArray}
+    "A reference to the collection to iterate over."
+    data::T
+    "The width of the sliding window."
+    width::Int
+    "How many positions the sliding window moves along."
+    step::Int
+    "The number of windows that will result from iterating across the container."
+    nwin::Int
+    "The number of elements in the container that are missed. Typically because
+    of the width and step size of the window: some elements near the end of the
+    container are missed if window sizes and step sizes are too large."
+    nmissed::Int
+end
```

Comment:
> The Julia documentation says you can attach docstrings to fields:

Wow! [You're right!](http://julia.readthedocs.org/en/latest/manual/documentation/#types).

> I thought nmissed would be useful for some purposes. In the past I've wanted to know how many positions are not covered by a sliding window run before. Likewise, nwin might be useful to know in advance how many windows are going to be generated.

Agreed, but in general I prefer "minimal fields" to avoid complexity. Since these values can be computed on demand with cheap cost, I think creating functions for such values would be a better approach.